bowtie/_report.py

```diff
@@ -288,6 +294,57 @@ def empty(
             did_fail_fast=False,
         )
 
+    @classmethod
+    def combine_versioned_reports_for(
+        cls,
+        versioned_reports: Iterable[Report],
+        dialect: Dialect,
+    ) -> Report | None:
+        versioned_reports = [
+            versioned_report
+            for versioned_report in versioned_reports
+            if versioned_report.metadata.dialect == dialect
+        ]
+
+        if versioned_reports:
+
+            cases: HashTrieMap[Seq, TestCase] = HashTrieMap()
+            results: HashTrieMap[
+                ConnectableId,
+                HashTrieMap[Seq, SeqResult],
+            ] = HashTrieMap()
+            implementations: dict[ConnectableId, ImplementationInfo] = {}
+
+            for versioned_report in versioned_reports:
+                for seq, case in versioned_report._cases.items():
+                    existing_case = cases.get(seq)
+                    if not existing_case:
+                        cases = cases.insert(seq, case)
+                    elif case != existing_case:
+                        raise CaseMismatchInReportsForSameDialect()
+
+                version_id, version_results = next(
```

Comment:
This line is a bit strange -- I'm not 100% sure what it is meant to do yet without trying to run this -- you're trying to get the first -- and just the first -- element from `_results`? Why?

Reply:
Because there would be just one in for e.g. https://bowtie.report/implementations/ruby-json_schemer/v2.3.0/draft7.json
while trying to combine in the for loop 😅.

Reply:
So is the length *always* 1?

Reply:
This method is essentially just trying to combine

https://bowtie.report/implementations/ruby-json_schemer/v2.3.0/draft7.json
https://bowtie.report/implementations/ruby-json_schemer/v2.2.1/draft7.json
https://bowtie.report/implementations/ruby-json_schemer/v2.2.0/draft7.json
..
..
..

into a single report, so yes, in the for loop for every report we always just have a single  _versioned_ implementation and results for it. And the cases are essentially the same for all of them i.e. of `draft7`. So I might as well just do `cls(cases=reports[0]._cases)` and entirely avoid `CaseMismatchInReportsForSameDialect`. Is it fine this way ?

Reply:
It's still hard to really follow here without dropping into a debugger -- please add a test or two for this functionality, which at least will make it a bit clearer, even though I think I follow what this is doing (it's assuming that all combined reports are for the same implementation, different versions, and that each input has one and only one version?)

Reply:
> it's assuming that all combined reports are for the same implementation, different versions, and that each input has one and only one version?

Yes!

Reply:
(the tests should/can probably go to `test_report.py` as I think they won't need to be integration tests?)

Reply:
Added. Can you have a look.